scripts/maintenance/computational-clusters/autoscaled_monitor/cli.py

```diff
@@ -46,19 +48,41 @@ def _parse_environment(deploy_config: Path) -> dict[str, str | None]:
     return environment
 
 
+def _parse_inventory(deploy_config: Path) -> BastionHost:
+    inventory_path = deploy_config / "ansible" / "inventory.ini"
+    if not inventory_path.exists():
+        rich.print(
+            f"[red]{inventory_path} does not exist! Please run `make inventory` in {deploy_config} to generate it[/red]"
+        )
+        raise typer.Exit(1)
+
+    loader = DataLoader()
+    inventory = InventoryManager(loader=loader, sources=[f"{inventory_path}"])
+
+    try:
+        return BastionHost(
+            ip=inventory.groups["CAULDRON_UNIX"].get_vars()["bastion_ip"],
+            user_name=inventory.groups["CAULDRON_UNIX"].get_vars()["bastion_user"],
+        )
+    except KeyError as err:
+        msg = "Unable to find bastion_ip in the inventory file."
+        raise RuntimeError(msg) from err
```

Comment:
Q: is this raised exception stopping the CLI execution. If yes, why not use type.Exit instead?

Reply:
:+1: